Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/AWSClientConfigDefaultsProvider.swift

```diff
@@ -99,17 +99,30 @@ public class AWSClientConfigDefaultsProvider {
         return resolvedRetryMode!
```

Comment:
GH diff is being a bit wonky here. I just added `private static func maxAttempts`, and changed the function signature of `func retryStrategyOptions` to take in arguments (the top level config `awsRetryMode` and `maxAttempts`) at call site.